framework/test/unit/modules/interoperability/method.spec.ts

```diff
@@ -448,4 +475,46 @@ describe('Sample Method', () => {
 			expect(channelStoreMock.get).toHaveBeenCalledWith(expect.anything(), newChainID);
 		});
 	});
+
+	describe('terminateChain', () => {
+		const sidechainChainAccount = {
+			name: 'sidechain1',
+			chainID: Buffer.alloc(CHAIN_ID_LENGTH),
+			lastCertificate: {
+				height: 10,
+				stateRoot: utils.getRandomBytes(32),
+				timestamp: 100,
+				validatorsHash: utils.getRandomBytes(32),
+			},
+			status: ChainStatus.TERMINATED,
+		};
+
+		beforeEach(() => {
+			interopMod['internalMethod'].sendInternal = jest.fn();
+			interopMod['internalMethod'].createTerminatedStateAccount = jest.fn();
+		});
+
+		it('should do nothing if chain was already terminated', async () => {
+			jest
+				.spyOn(sampleInteroperabilityMethod as any, 'getTerminatedStateAccount')
+				.mockResolvedValue({
+					stateRoot: sidechainChainAccount.lastCertificate.stateRoot,
+					mainchainStateRoot: Buffer.alloc(HASH_LENGTH),
+					initialized: true,
+				});
+			await sampleInteroperabilityMethod.terminateChain(terminateChainContext, chainID);
+
+			expect(interopMod['internalMethod'].sendInternal).not.toHaveBeenCalled();
+		});
+
+		it('should process with input chainID', async () => {
+			jest
+				.spyOn(sampleInteroperabilityMethod as any, 'getTerminatedStateAccount')
+				.mockResolvedValue(null);
+
+			await sampleInteroperabilityMethod.terminateChain(terminateChainContext, chainID);
+			expect(interopMod['internalMethod'].sendInternal).toHaveBeenCalledTimes(1);
```

Comment:
maybe also check with `toHaveBeenCalledWith` ? for both `sendInternal` and `createTerminatedStateAccount `